spec/models/spree/ability_spec.rb

```diff
@@ -514,5 +514,89 @@ module Spree
+      let!(:product) { create(:product, supplier: producer) }
+      let!(:variant) { create(:variant, product: product) }
+      let!(:variant_override) { create(:variant_override, hub: distributor, variant: variant) }
+
+      subject { user }
+
+      let(:manage_actions) { [:admin, :index, :read, :update, :bulk_update, :bulk_reset] }
+
+      describe "when admin" do
+        let(:user) { create(:admin_user) }
+
+        it "should have permission" do
+          should have_ability(manage_actions, for: variant_override)
+        end
+      end
+
+      describe "when user of the producer" do
+        let(:user) { producer.owner }
+
+        it "should not have permission" do
+          should_not have_ability(manage_actions, for: variant_override)
+        end
+      end
+
+      describe "when user of the distributor" do
+        let(:user) { distributor.owner }
+
+        it "should not have permission" do
+          should_not have_ability(manage_actions, for: variant_override)
+        end
+      end
+
+      describe "when user of the distributor which is also the producer" do
+        let(:user) { distributor.owner }
+        let!(:distributor) { create(:distributor_enterprise, is_primary_producer: true, sells: "any") }
+        let!(:producer) { distributor }
+
+        it "should have permission" do
+          should have_ability(manage_actions, for: variant_override)
+        end
+      end
+
+      describe "when owner of the distributor with add_to_order_cycle permission to the producer" do
+        let!(:unauthorized_enterprise) do
+          create(:enterprise, sells: "any").tap do |record|
+            create(:enterprise_relationship, parent: producer, child: record, permissions_list: [:add_to_order_cycle])
+          end
+        end
+        let(:user) { unauthorized_enterprise.owner }
+
+        it "should not have permission" do
+          should_not have_ability(manage_actions, for: variant_override)
+        end
+      end
+
+      describe "when owner of the enterprise with create_variant_overrides permission to the producer" do
+        let!(:authorized_enterprise) do
+          create(:enterprise, sells: "any").tap do |record|
+            create(:enterprise_relationship, parent: producer, child: record, permissions_list: [:create_variant_overrides])
+          end
+        end
+        let(:user) { authorized_enterprise.owner }
+
+        it "should not have permission" do
+          should_not have_ability(manage_actions, for: variant_override)
+        end
+
+        describe "when the enterprise is not a distributor" do
+          let!(:authorized_enterprise) do
+            create(:enterprise, sells: "none").tap do |record|
+              create(:enterprise_relationship, parent: producer, child: record, permissions_list: [:create_variant_overrides])
+            end
+          end
+
+          it "should not have permission" do
+            should_not have_ability(manage_actions, for: variant_override)
```

Comment:
Could we use RSpec-3 syntax here? I know I sound boring and stubborn but sticking to this very old syntax makes it hard to stay-up-to-date. Let me know if you need help.

I'd like to automate this so I can avoid having to review syntax but I don't really know how to. Disabling `should` as RSpec's config allows is an all-or-nothing.

Reply:
ok, as you guys want. It's just that if we don't all get used to not writing RSpec2 syntax it'll make it harder.

Reply:
I rebased the branch and also changed to RSpec 2 syntax.

@sauloperez Sorry about that! This was an old PR, before I found out about `is_expected` from you. :slightly_smiling_face: Odd but I insisted on using `should` when dealing with `subject` until that time.

@luisramos0 I changed only the syntax in the tests I added, so no extra problems when merging.

Reply:
:+1: sorry, maybe I was too quick, my "no transpec until 2-0 merge" is not really valid for new tests...